### PR TITLE
[XLA] AlgebraicSimplifier pass - handle And and Or

### PR DESCRIPTION
**Pull request by georgepaw**
Add handling of logical And and Or, where:
* (a && True) => a
* (True && a) => a
* (a && False) => False
* (False && a) => False
* (a || True) => True
* (True || a) => True
* (a || False) => a
* (False || a) => a

Added tests for these as well.

### CODE REVIEW

File: tensorflow/compiler/xla/service/algebraic_simplifier.cc [hunk 1]
```diff
@@ -419,6 +423,45 @@ Status AlgebraicSimplifierVisitor::HandleAdd(HloInstruction* add) {
...
+    }
+
+    // False && A => False
+    VLOG(10) << "trying transform [False && A => False]: "
+             << logical_and->ToString();
+    if (IsAll(lhs, 0) && ReplaceInstructionIfSameShape(logical_and, lhs)) {
+      return Status::OK();
+    }
+
+    return Status::OK();
```
Comment (congliuthu) [inline tensorflow/compiler/xla/service/algebraic_simplifier.cc]: Nit: this return can be removed.

File: tensorflow/compiler/xla/service/algebraic_simplifier.cc [hunk 2]
```diff
@@ -1225,6 +1268,46 @@ Status AlgebraicSimplifierVisitor::HandleMultiply(HloInstruction* multiply) {
...
+    }
+
+    // False || A => A
+    VLOG(10) << "trying transform [False || A => A]: "
+             << logical_or->ToString();
+    if (IsAll(lhs, 0) && ReplaceInstructionIfSameShape(logical_or, rhs)) {
+      return Status::OK();
+    }
+
+    return Status::OK();
```
Comment (congliuthu) [inline tensorflow/compiler/xla/service/algebraic_simplifier.cc]: Same here.

**Review by congliuthu — DISMISSED** [review]
Looks good % minor nits.

File: tensorflow/compiler/xla/service/algebraic_simplifier.cc (hunk 1 above)
Comment (georgepaw) [reply tensorflow/compiler/xla/service/algebraic_simplifier.cc]: Done.

File: tensorflow/compiler/xla/service/algebraic_simplifier.cc (hunk 2 above)
Comment (georgepaw) [reply tensorflow/compiler/xla/service/algebraic_simplifier.cc]: Done.

**Comment by georgepaw** [comment]
Thank you for the review, all done now

**Review by congliuthu — APPROVED** [review]
Thanks!

**Pull request merged**